source/simulator/stokes_matrix_free.cc

```diff
@@ -450,14 +450,15 @@ namespace aspect
          * @param S_block_tolerance The tolerance for the CG solver which computes
          *     the inverse of the S block (Schur complement matrix).
          */
-        BlockSchurGMGPreconditioner (const StokesMatrixType  &S,
-                                     const ABlockMatrixType  &A,
-                                     const MassMatrixType  &Mass,
-                                     const PreconditionerMp                     &Mppreconditioner,
-                                     const PreconditionerA                      &Apreconditioner,
-                                     const bool                                  do_solve_A,
-                                     const double                                A_block_tolerance,
-                                     const double                                S_block_tolerance);
+        BlockSchurGMGPreconditioner (const StokesMatrixType &S,
```

Comment:
The parameter names for this function are somewhat confusing (not only since your addition). Can we maybe reorganize so that `S = Stokes` is distinguishable from `S = SchurComplement`. Also we currently implicitly assume that the Schur complement is a mass matrix, but that might not stay like this. Would the following parameter be more reasonable (also asking @tjhei)?
```
        BlockSchurGMGPreconditioner (const StokesMatrixType &Stokes_matrix,
                                     const ABlockMatrixType &A_block,
                                     const MassMatrixType   &Schur_complement_block,
                                     const PreconditionerA  &A_block_preconditioner,
                                     const PreconditionerMp &Schur_complement_preconditioner,
                                     const bool              do_solve_A_block,
                                     const bool              do_solve_Schur_complement,
                                     const double            A_block_tolerance,
                                     const double            Schur_complement_tolerance);

Reply:
Looks reasonable to me. We can also change the template name from ``MassMatrixType`` to ``SchurComplementMatrixType`` or something. Similarly, ``PreconditionerMp``.

Reply:
I agree with the renaming as well. I've changed many names and the documentation of the  `BlockSchurGMGPreconditioner` class, as well as a few names in the `StokesMatrixFreeHandlerImplementation` class  (e.g., `_M` in most places becomes `_Schur`).

I still keep the naming `MassMatrixOperator` for the matrix-free class that describes the pressure mass matrix.